internal/services/namedlocations/named_location_resource.go

```diff
@@ -0,0 +1,125 @@
+package namedlocations
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/tf"
+	"github.com/hashicorp/terraform-provider-azuread/internal/validate"
+)
+
+func namedLocationResource() *schema.Resource {
+	eactlyOneOf := []string{"ip", "country"}
+	return &schema.Resource{
+		CreateContext: namedLocationResourceCreate,
+		ReadContext:   namedLocationResourceRead,
+		UpdateContext: namedLocationResourceUpdate,
+		DeleteContext: namedLocationResourceDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Importer: tf.ValidateResourceIDPriorToImport(func(id string) error {
+			if _, err := uuid.ParseUUID(id); err != nil {
+				return fmt.Errorf("specified ID (%q) is not valid: %s", id, err)
+			}
+			return nil
+		}),
+
+		Schema: map[string]*schema.Schema{
+
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
We don't need to include `id` in the schema, the plugin SDK manages it implicitly.

Reply:
Noted and removed

Reply:
@manicminer tests fail without this as a result of panic: Invalid address to set: []string{"id"} from what I can see- Unless the branch needs updating with the latest sdk

Reply:
Ah I see, you need to use the `d.SetId()` method rather than `d.Set("id")` / `tf.Set(d, "id")`

Reply:
Ah thanks, that makes more sense now